pkg/ruler/manager.go

```diff
@@ -331,6 +331,11 @@ func (*DefaultMultiTenantManager) ValidateRuleGroup(g rulefmt.RuleGroup) []error
 		return errs
 	}
 
+	if !r.cfg.TenantFederation.Enabled && len(g.SourceTenants) > 0 {
+		errs = append(errs, fmt.Errorf("invalid rules config: rule group '%s' is a federated rule group but rules federation is disabled. "+
+			"To enable it, set -ruler.tenant-federation.enabled=true as a CLI argument or ruler.tenant_federation.enabled: true in YAML or contact your service administrator", g.Name))
```

Comment:
[nit] WDYT about defining `"ruler.tenant-federation.enabled"` as a const so that we reuse it both here and in the CLI flag definition?